src/token_swapper.rs

```diff
@@ -0,0 +1,61 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use crate::graph;
+use crate::iterators::EdgeList;
+
+use hashbrown::HashMap;
+use petgraph::graph::NodeIndex;
+use pyo3::prelude::*;
+use rustworkx_core::token_swapper;
+
+/// This module performs an approximately optimal Token Swapping algorithm
+/// Supports partial mappings (i.e. not-permutations) for graphs with missing tokens.
+///
+/// Based on the paper: Approximation and Hardness for Token Swapping by Miltzow et al. (2016)
+/// ArXiV: https://arxiv.org/abs/1602.05150
+/// and generalization based on our own work.
```

Comment:
```suggestion
```

Reply:
Done in https://github.com/Qiskit/rustworkx/pull/765/commits/7a7ae3bca4873c0a7166cd005289bde33470de39.